x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/store/SearchableSnapshotDirectoryTests.java

```diff
@@ -786,7 +787,12 @@ public void testClearCache() throws Exception {
                     }
                     assertListOfFiles(cacheDir, allOf(greaterThan(0), lessThanOrEqualTo(nbRandomFiles)), greaterThan(0L));
                     if (randomBoolean()) {
-                        directory.clearCache();
+                        directory.clearCache(); // should not clear CacheService cache
+                        assertBusy(() -> assertListOfFiles(cacheDir, greaterThan(0), greaterThan(0L)));
```

Comment:
See previous comment, I'd use `directory.clearCaches(true, true)` here and avoid an extra assertBusy